psasim/test/Makefile

```diff
@@ -20,7 +20,10 @@ psa_partition: $(GENERATED_H_FILES)
 psa_partition: psa_ff_bootstrap_TEST_PARTITION.c
 	$(CC) $(INCLUDE) $(CFLAGS) $< $(LIB) -o $@
 
-psa_ff_bootstrap_%.c psa_manifest/%.h: server.c
+psa_ff_bootstrap_%.c: server.c
```

Comment:
The `%` here is really being used as a wildcard/shortcut cheat, rather than as part of an pattern rule (where there would be something like `%.o : %.c`). In that case, and since we've already got an instance of the `psa_ff_bootstrap_%.c` pattern written out a few lines above, maybe we should define a variable for `psa_ff_bootstrap_TEST_PARTITION.c` and use it together with `$(GENERATED_H_FILES)` to not repeat the `psa_autogen.py` recipe.

What do you think? (And sorry this is taking longer than expected)

Reply:
So we've then got something like

```
PARTITION_SERVER_BOOTSTRAP = psa_ff_bootstrap_TEST_PARTITION.c

psa_client: client.c $(GENERATED_H_FILES)
	$(CC) $(INCLUDE) $(CFLAGS) client.c $(LIB) -o $@

psa_partition: $(PARTITION_SERVER_BOOTSTRAP) $(GENERATED_H_FILES)
	$(CC) $(INCLUDE) $(CFLAGS) $< $(LIB) -o $@

$(PARTITION_SERVER_BOOTSTRAP) $(GENERATED_H_FILES): server.c
	../tools/psa_autogen.py manifest.json
```

Reply:
> And sorry this is taking longer than expected

No worries at all!! Reviews are meant to improve code and this is exactly what it is happening here :)
Thanks for the tips